src/machine/machine-varlink.c

```diff
@@ -188,3 +188,99 @@ int vl_method_register(sd_varlink *link, sd_json_variant *parameters, sd_varlink
 
         return sd_varlink_reply(link, NULL);
 }
+
+static int lookup_machine_by_name(sd_varlink *link, Manager *manager, const char *machine_name, Machine **ret_machine) {
+        assert(link);
+        assert(manager);
+        assert(ret_machine);
+
+        if (!machine_name)
+                return -EINVAL;
+
+        if (!hostname_is_valid(machine_name, /* flags= */ VALID_HOSTNAME_DOT_HOST))
+                return -EINVAL;
+
+        Machine *machine = hashmap_get(manager->machines, machine_name);
+        if (!machine)
+                return -ESRCH;
```

Comment:
Some debug logging on errors here will likely be useful

Reply:
going to ignore it as agreed